src/coreclr/inc/corinfo.h

```diff
@@ -850,6 +850,7 @@ enum CorInfoFlag
 //  CORINFO_FLG_UNUSED                = 0x40000000,
 
     // These are internal flags that can only be on Classes
+    CORINFO_FLG_SPAN                  = 0x00008000, // is the class a Span<T>/ReadOnlySpan<T>
```

Comment:
The default pattern for special casing types in the JIT is to mark the type with `[Intrinsic]`, call `isIntrinsicType` on the class handle, and check the name if `isIntrinsicType` returns true. It is how we are special casing the SIMD Vector types. My guess that the default pattern would cheaper overall than computing all type flags just to figure out whether the type is a Span or ReadOnlySpan.

If this check is very hot and we cannot afford to use the default pattern for it, it should be a dedicated `isSpanOrReadOnlySpanType` method on JIT/EE interface so that we do not pay the full cost of getting all flags that `getClassAttributes` returns.

Reply:
I suggested to @En3Tho to add `CORINFO_FLG_SPAN` mainly based on the fact that it is on the same level of information as `CORINFO_FLG_ARRAY` and `CORINFO_FLG_DELEGATE`,  but it is reasonable to me to switch this to `isSpanOrReadOnlySpanType`. I am not so sure that we want to switch `lvaSetStruct` to be doing the same style of recognition as the SIMD recognition we do today.

> > My guess that the default pattern would cheaper overall than computing all type flags just to figure out whether the type is a Span or ReadOnlySpan.

Note that we have been calling `getClassStribs` from `lvaSetStruct` since forever. It was only recently removed in #87967.

Reply:
> CORINFO_FLG_SPAN mainly based on the fact that it is on the same level of information as CORINFO_FLG_ARRAY and CORINFO_FLG_DELEGATE

Nit: Arrays or Delegates are core type system entities that one cannot identify via a simple name check. Span/ReadOnlySpan is just an ordinary type that the JIT is recognizing as intrinsic. 

> I am not so sure that we want to switch lvaSetStruct to be doing the same style of recognition as the SIMD recognition we do today.

Are you worried about throughput? The name check behind the `isIntrinsicType` early out is fairly cheap overall. We have intentionally switched to allow JIT to identify intrinsics by name, so that the JIT/EE interface does not need to know about all ordinary types and methods that the JIT wants to special case as an intrinsic.

If we are worried about throughout, a dedicated `isSpanOrReadOnlySpanType` method would be the way to go.

Reply:
> Nit: Arrays or Delegates are core type system entities that one cannot identify via a simple name check. Span/ReadOnlySpan is just an ordinary type that the JIT is recognizing as intrinsic.

Fair point.

> Are you worried about throughput? The name check behind the isIntrinsicType early out is fairly cheap overall. We have intentionally switched to allow JIT to identify intrinsics by name, so that the JIT/EE interface does not need to know about all ordinary types and methods that the JIT wants to special case as an intrinsic.

Yes, I was mainly worried about throughput -- we have been quite careful in the JIT to only do SIMD recognition from the paths where we know it is relevant, instead of, say, a general function like `lvaSetStruct`. But SIMD recognition also has much more to recognize, so if you say that the `isIntrinsic` + metadata lookup is cheap, then I think we should just do it this way and avoid the JIT-EE change.

Reply:
We can keep track of it only when optimizing.